Source/ablastr/fields/IntegratedGreenFunctionSolver.cpp

```diff
@@ -212,81 +228,156 @@ computePhiIGF ( amrex::MultiFab const & rho,
         tmp_rho_fft.shift(realspace_box.smallEnd());
         tmp_G_fft.shift(realspace_box.smallEnd());
 
-        // Create FFT plans
-        BL_PROFILE_VAR_START(timer_plans);
-#if !defined(ABLASTR_USE_HEFFTE)
-        const amrex::IntVect fft_size = realspace_ba[local_boxid].length();
-        ablastr::math::anyfft::FFTplan forward_plan_rho = ablastr::math::anyfft::CreatePlan(
-            fft_size, tmp_rho[local_boxid].dataPtr(),
-            reinterpret_cast<ablastr::math::anyfft::Complex*>(tmp_rho_fft.dataPtr()),
-            ablastr::math::anyfft::direction::R2C, AMREX_SPACEDIM);
-        ablastr::math::anyfft::FFTplan forward_plan_G = ablastr::math::anyfft::CreatePlan(
-            fft_size, tmp_G[local_boxid].dataPtr(),
-            reinterpret_cast<ablastr::math::anyfft::Complex*>(tmp_G_fft.dataPtr()),
-            ablastr::math::anyfft::direction::R2C, AMREX_SPACEDIM);
-        ablastr::math::anyfft::FFTplan backward_plan = ablastr::math::anyfft::CreatePlan(
-            fft_size, tmp_G[local_boxid].dataPtr(),
-            reinterpret_cast<ablastr::math::anyfft::Complex*>( tmp_G_fft.dataPtr()),
-            ablastr::math::anyfft::direction::C2R, AMREX_SPACEDIM);
-#elif defined(ABLASTR_USE_HEFFTE)
-#if     defined(AMREX_USE_CUDA)
-        heffte::fft3d_r2c<heffte::backend::cufft> fft
-#elif   defined(AMREX_USE_HIP)
-        heffte::fft3d_r2c<heffte::backend::rocfft> fft
+        // Number of cells in real space
+        const int nrx = local_box.length(0);
+        const int nry = local_box.length(1);
+        const int nrz = local_box.length(2);
+
+        // Number of cells in spectral space
+        const int nsx = c_local_box.length(0);
+        const int nsy = c_local_box.length(1);
+        const int nsz = c_local_box.length(2);
+
+        if(is_2d_slices){
+                        // Create many plans at once
+            int fft_size[] = {nry, nrx};
+            BL_PROFILE_VAR_START(timer_plans);
+            ablastr::math::anyfft::FFTplan forward_plan_rho = ablastr::math::anyfft::CreatePlanMany(
+                fft_size, tmp_rho[local_boxid].dataPtr(),
+                reinterpret_cast<ablastr::math::anyfft::Complex*>(tmp_rho_fft.dataPtr()),
+                ablastr::math::anyfft::direction::R2C, AMREX_SPACEDIM-1,
+                nrz, nullptr, 1, nrx*nry, nullptr, 1, nsx*nsy);
```

Comment:
Could we modify the interface of `CreatePlanMany` so that the user does not need to pass `nullptr, 1` ?

Reply:
We could, but I wanted to keep the wrapper as flexible and general as possible.
What if one day anyone will need batched ffts that use those additional parameters?
That was my idea, but I can make the `CreatePlanMany` less flexible, if you think it's best.